### PR TITLE
Removed deprecated Mail methods

### LINKED ISSUE #12924: [Mail] Remove deprecated methods
### Feature description

- https://github.com/pimcore/pimcore/blob/27b0b2201e44e6da9f69f80ecf87bcee7840ab26/lib/Mail.php#L879
- https://github.com/pimcore/pimcore/blob/27b0b2201e44e6da9f69f80ecf87bcee7840ab26/lib/Mail.php#L866
- https://github.com/pimcore/pimcore/blob/27b0b2201e44e6da9f69f80ecf87bcee7840ab26/lib/Mail.php#L912
- https://github.com/pimcore/pimcore/blob/dedb59336dabf4ce5288b514dcffeb82d34a6eaf/lib/Mail.php#L966

### PR DESCRIPTION
**Pull request by aryaantony92**
## Changes in this pull request  
Resolves #12924

### CODE REVIEW

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 1]
```diff
@@ -29,11 +29,30 @@
...
+  After:
+    ```php
+        $mail= new \Pimcore\Mail($headers = null, $body = null, $contentType = null);
+        $mail->html("<b>some</b> rich text: {{ myParam }}");
+        $mail->text("This is just plain text");
+        $mail->attach($asset->getData(), $asset->getFilename(), $asset->getMimeType());
+        $mail->subject('Test Mail');
+        ...
+    ```
+  For details, please see [#12924] (https://github.com/pimcore/pimcore/issues/12924)
```
Comment (dvesh3) [inline doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]:
```suggestion
```
No need of examples here. One liner about removal is fine.

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 2]
```diff
@@ -30,9 +30,11 @@
 - [Elements] Removed fallback to parent id 1, when an element with a non-existing parent id gets created.
 - [DataObjects] Added return types to setter methods. For details see [#12185](https://github.com/pimcore/pimcore/issues/12185)
 - [Exception] Deprecated MissingDependencyException has been removed.
-- [CustomLayouts] Removed command `pimcore:deployment:custom-layouts-rebuild` as CustomLayouts are migrated to LocationAwareConfigRepository.
+- [CustomLayouts] Removed command `pimcore:deployment:custom-layouts-rebuild` as CustomLayouts are migrated to LocationAwareConfigRepository.<<<<<<< remove_deprecated_mail_methods
```
Comment (dvesh3) [inline doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]:
```suggestion
- [CustomLayouts] Removed command `pimcore:deployment:custom-layouts-rebuild` as CustomLayouts are migrated to LocationAwareConfigRepository.
```

Comment (aryaantony92) [reply doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]: Already committed now.

**Review by aashan10 — APPROVED** [review]

**Comment by aashan10** [comment]
@aryaantony92 please fix the merge conflicts.

**Comment by aryaantony92** [comment]
> @aryaantony92 please fix the merge conflicts.

Done 👍

**Pull request merged**